doc/examples/controls/file_download-dynamic.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+# -----------------------------------------------------------------------------------------
+# To execute this script, make sure that the taipy-gui package is installed in your
+# Python environment and run:
+#     python <script>
+# -----------------------------------------------------------------------------------------
+from taipy.gui import Gui, download
+from decimal import getcontext, Decimal
+import io
+
+# Initial precision
+precision = 10
+
+
+def pi(precision: int) -> list[int]:
+    """Compute Pi to the required precision.
+
+    Adapted from https://docs.python.org/3/library/decimal.html
+    """
+    saved_precision = getcontext().prec  # Save precision
+    getcontext().prec = precision
+    three = Decimal(3)  # substitute "three=3.0" for regular floats
+    lasts, t, s, n, na, d, da = 0, three, 3, 1, 0, 0, 24
+    while s != lasts:
+        lasts = s
+        n, na = n + na, na + 8
```

Comment:
this is not really easy to read ?

Reply:
That comes straight from the Python doc. I won't change that.